File: Plugins/Aspose_Pdf_Java_for_Ruby/lib/asposepdfjava/Text/replacetextallpages.rb
module Asposepdfjava
  module ReplaceTextAllPages
    def initialize()
    	# The path to the documents directory.
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Open the target document
        pdf_document = Rjb::import('com.aspose.pdf.Document').new(data_dir + 'input.pdf')

        # Create TextAbsorber object to find all instances of the input search phrase
        text_fragment_absorber = Rjb::import('com.aspose.pdf.TextFragmentAbsorber').new("sample")

        # Accept the absorber for first page of document
        pdf_document.getPages().accept(text_fragment_absorber)

        # Get the extracted text fragments into collection
        text_fragment_collection = text_fragment_absorber.getTextFragments()
        
        color = Rjb::import('com.aspose.pdf.Color')
        font_repository = Rjb::import('com.aspose.pdf.FontRepository')
        text_fragment = Rjb::import('com.aspose.pdf.TextFragment')
        
        # Loop through the fragments
        #for(text_fragment : (Iterable<com.aspose.pdf.TextFragment>)textFragmentCollection)
        i = 0
        while i < text_fragment_collection
            # Update text and other properties
            text_fragment.setText("New Phrase")
            text_fragment.getTextState().setFont(font_repository.findFont("Verdana"))
            text_fragment.getTextState().setFontSize(22)
            text_fragment.getTextState().setForegroundColor(color.getBlue())
            text_fragment.getTextState().setBackgroundColor(color.getGray())
            i +=1
        end
        # Save the updated PDF file
        pdfDocument.save(data_dir+ "ReplaceTextAllPages.pdf")
    end
  end
end

File: Examples/src/main/java/com/aspose/pdf/examples/AsposePdf/Text/ReplaceTextAllPages.java
/* 
 * Copyright 2001-2015 Aspose Pty Ltd. All Rights Reserved.
 *
 * This file is part of Aspose.Pdf. The source code in this file
 * is only intended as a supplement to the documentation, and is provided
 * "as is", without warranty of any kind, either expressed or implied.
 */
 
package com.aspose.pdf.examples.AsposePdf.Text;

import com.aspose.pdf.examples.Utils;


public class ReplaceTextAllPages
{
    public static void main(String[] args) throws Exception
    {
        // The path to the documents directory.
        String dataDir = Utils.getDataDir(ReplaceTextAllPages.class);

        // Open document
        com.aspose.pdf.Document pdfDocument = new com.aspose.pdf.Document(dataDir+ "input.pdf");

        // Create TextAbsorber object to find all instances of the input search phrase
        com.aspose.pdf.TextFragmentAbsorber textFragmentAbsorber = new com.aspose.pdf.TextFragmentAbsorber("sample");

        // Accept the absorber for first page of document
        pdfDocument.getPages().accept(textFragmentAbsorber);

        // Get the extracted text fragments into collection
        com.aspose.pdf.TextFragmentCollection textFragmentCollection = textFragmentAbsorber.getTextFragments();

        // Loop through the fragments
        for(com.aspose.pdf.TextFragment textFragment : (Iterable<com.aspose.pdf.TextFragment>)textFragmentCollection)
        {
            // Update text and other properties
            textFragment.setText("New Phrase");
            textFragment.getTextState().setFont(com.aspose.pdf.FontRepository.findFont("Verdana"));
            textFragment.getTextState().setFontSize(22);
            textFragment.getTextState().setForegroundColor(com.aspose.pdf.Color.getBlue());
            textFragment.getTextState().setBackgroundColor(com.aspose.pdf.Color.getGray());
        }
        // Save the updated PDF file
        pdfDocument.save(dataDir+ "output.pdf");

    }
}




